packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -74,35 +59,118 @@ export const Alert = React.forwardRef((props, ref) => {
       ref={ref}
       {...rest}
     >
-      <StatusIcon className='iui-alert-icon' />
-      <Box as='span' className='iui-alert-message'>
-        {children}
-        {clickableText && (
-          <Box
-            as='a'
-            {...clickableTextProps}
-            className={cx('iui-alert-link', clickableTextProps?.className)}
-          >
-            {clickableText}
-          </Box>
-        )}
-      </Box>
-
-      {onClose && (
-        <Box
-          as='button'
-          onClick={onClose}
-          aria-label='Close'
-          type='button'
-          className='iui-alert-button'
-        >
-          <Box as='span' className='iui-alert-button-icon' aria-hidden>
-            <SvgCloseSmall />
-          </Box>
-        </Box>
-      )}
+      <AlertContext.Provider value={{ type }}>{children}</AlertContext.Provider>
     </Box>
   );
-}) as PolymorphicForwardRefComponent<'div', AlertProps>;
+}) as PolymorphicForwardRefComponent<'div', AlertOwnProps>;
+AlertComponent.displayName = 'Alert';
+
+// ----------------------------------------------------------------------------
+// Alert.Icon component
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+
+  const { type } = useSafeContext(AlertContext);
+
+  const StatusIcon = StatusIconMap[type];
+
+  return (
+    <Icon fill={type} ref={ref} {...rest}>
+      {children ?? <StatusIcon />}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', React.ComponentProps<typeof Icon>>;
+AlertIcon.displayName = 'Alert.Icon';
+
+// ----------------------------------------------------------------------------
+// Alert.Message component
+
+const AlertMessage = polymorphic.span('iui-alert-message');
+AlertMessage.displayName = 'Alert.Message';
+
+// ----------------------------------------------------------------------------
+// Alert.Action component
+
+const AlertAction = React.forwardRef((props, ref) => {
+  const { children, className, ...rest } = props;
+
+  return (
+    <Box
+      as={(!!props.href ? 'a' : 'button') as 'a'}
```

Comment:
We should have unit test for this.

Reply:
Added unit test

Reply:
Also test if it changes to <a> when href provided.

Reply:
I have added a selector for `a` when getting the `link` variable in one of the unit tests